robot_smach_states/test/navigation/test_topological_navigation.py

```diff
@@ -0,0 +1,72 @@
+# System
+import mock
+import unittest
+
+# TU/e Robotics
+from topological_action_planner_msgs.msg import Edge, Node
+
+# Robot Smach States
+from robot_smach_states.manipulation.open_door import PassDoor
+from robot_smach_states.navigation.navigate_to_symbolic import NavigateToSymbolic
+from robot_smach_states.navigation.navigate_to_waypoint import NavigateToWaypoint
+from robot_smach_states.topological_navigation import TopologicalPlannerException, convert_msgs_to_actions
+
+SOURCE_NODE_ID = "SOURCE_NODE"
+SOURCE_NODE_AREA = "IN_FRONT_OF"
+DESTINATION_NODE_ID = "DESTINATION_NODE"
+DESTINATION_NODE_AREA = "IN_FRONT_OF"
+ROBOT = mock.Mock()
+
+
+class TestTopologicalNavigation(unittest.TestCase):
+    """
+    Tests the TopologicalNavigation smach state
+    """
+    def test_drive(self):
+        msg = Edge()
+        msg.action_type = Edge.ACTION_DRIVE
+        msg.origin = Node(SOURCE_NODE_ID, SOURCE_NODE_AREA)
+        msg.destination = Node(DESTINATION_NODE_ID, DESTINATION_NODE_AREA)
+        actions = convert_msgs_to_actions(ROBOT, [msg])
+        self.assertEquals(len(actions), 1, "Result should contain exactly one action")
+        self.assertTrue(isinstance(actions[0], NavigateToSymbolic), "Action should be a 'NavigateToSymbolic' action")
+
+    def test_drive_waypoint(self):
+        msg = Edge()
+        msg.action_type = Edge.ACTION_DRIVE
+        msg.origin = Node(SOURCE_NODE_ID, SOURCE_NODE_AREA)
+        msg.destination.entity = DESTINATION_NODE_ID
+        actions = convert_msgs_to_actions(ROBOT, [msg])
+        self.assertEquals(len(actions), 1, "Result should contain exactly one action")
+        self.assertTrue(isinstance(actions[0], NavigateToWaypoint), "Action should be a 'NavigateToWaypoint' action")
+
+    def test_open_door(self):
+        msg = Edge()
+        msg.action_type = Edge.ACTION_OPEN_DOOR
+        entity_id = "door"
+        msg.origin = Node(entity_id, SOURCE_NODE_AREA)
+        msg.destination = Node(entity_id, DESTINATION_NODE_AREA)
+        actions = convert_msgs_to_actions(ROBOT, [msg])
+        self.assertEquals(len(actions), 1, "Result should contain exactly one action")
+        self.assertTrue(isinstance(actions[0], PassDoor), "Action should be a 'OpenDoor' action")
+
+    def test_open_door_wrong_entity(self):
+        msg = Edge()
+        msg.action_type = Edge.ACTION_USE_ELEVATOR
```

Comment:
This test seems to be testing how the system reacts to an unknown action type. That is not apparent from the name of the test however.